### PR TITLE
{AppService} az webapp config ssl import: Fix to allow key vault in other subscription

### PR DESCRIPTION
**Pull request by madsd**
**History Notes:**  

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by arrownj** [comment]
Hi @qwordy @Juliehzl , could you please help review this PR ?

**Comment by yonzhan** [comment]
add to S165.

**Comment by qwordy** [comment]
What's the disadvantage of allowing key vault in other subscription? Was it supported before?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -2054,10 +2055,15 @@ def import_ssl_cert(cmd, resource_group_name, name, key_vault, key_vault_certifi
...
-def _check_service_principal_permissions(cmd, resource_group_name, key_vault_name):
+def _check_service_principal_permissions(cmd, resource_group_name, key_vault_name, key_vault_subscription):
     from azure.cli.command_modules.keyvault._client_factory import keyvault_client_vaults_factory
     from azure.cli.command_modules.role._client_factory import _graph_client_factory
     from azure.graphrbac.models import GraphErrorException
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    subscription = get_subscription_id(cmd.cli_ctx)
+    # Cannot check if key vault is in another subscription
+    if subscription != key_vault_subscription:
+        return False
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Your code looks good

**Comment by madsd** [comment]
> What's the disadvantage of allowing key vault in other subscription? Was it supported before?

It is allowed and is still supported - all this change does is remove the access check, as the key vault sdk does not allow cross subscription lookup.

**Comment by madsd** [comment]
@zhoxing-ms , @qianwens , @Juliehzl can you have a look at this. I just pushed with a fresh rebase. It is a minor change to allow use of Key Vaults from other subscriptions. The functionality was working, but the check for permissions did not, so essentially I just removed the check, if the subscriptions are different (and the call will then just fail in the backend as expected)

**Comment by madsd** [comment]
@yonzhan is there a reason that this is just pushed?

**Pull request merged**